src/compas_timber/ghpython/components/CT_Model/code.py

```diff
@@ -112,71 +115,73 @@ def get_joints_from_rules(self, beams, rules, topologies):
-                try:
-                    joint.joint_type.create(Model, *beams_to_pair, **joint.kwargs)
-                except BeamJoinningError as bje:
-                    debug_info.add_joint_error(bje)
-                else:
-                    handled_beams.append(beam_pair_ids)
+        if create_geometry:
+            self.model = TimberModel()
+            for beam in beams:
+                # prepare beams for downstream processing
+                beam.remove_features()
+                beam.remove_blank_extension()
+                beam.debug_info = []
+                self.model.add_beam(beam)
+            topologies = []
+            solver = ConnectionSolver()
+            found_pairs = solver.find_intersecting_pairs(beams, rtree=True, max_distance=max_distance)
+            for pair in found_pairs:
+                beam_a, beam_b = pair
+                detected_topo, beam_a, beam_b = solver.find_topology(beam_a, beam_b, max_distance=max_distance)
+                if not detected_topo == JointTopology.TOPO_UNKNOWN:
+                    topologies.append({"detected_topo": detected_topo, "beam_a": beam_a, "beam_b": beam_b})
+            self.model.set_topologies(topologies)
+
+            beams = self.model.beams
+            joints = self.get_joints_from_rules(beams, joint_rules, topologies)
+
+            if joints:
+                handled_beams = []
+                joints = [j for j in joints if j is not None]
+                # apply reversed. later joints in orginal list override ealier ones
+                for joint in joints[::-1]:
+                    beams_to_pair = joint.beams
+                    beam_pair_ids = set([id(beam) for beam in beams_to_pair])
+                    if beam_pair_ids in handled_beams:
+                        continue
+                    try:
+                        joint.joint_type.create(self.model, *beams_to_pair, **joint.kwargs)
+                    except BeamJoinningError as bje:
+                        debug_info.add_joint_error(bje)
+                    else:
+                        handled_beams.append(beam_pair_ids)
 
-        if Features:
-            features = [f for f in Features if f is not None]
-            for f_def in features:
-                for beam in f_def.beams:
-                    beam.add_features(f_def.feature)
+            if features:
+                features = [f for f in features if f is not None]
+                for f_def in features:
+                    for beam in f_def.beams:
+                        beam.add_features(f_def.feature)
 
-        Geometry = None
+        geometry = None
         scene = Scene()
-        for beam in Model.beams:
-            if CreateGeometry:
+        for beam in self.model.beams:
+            try:
                 scene.add(beam.geometry)
                 if beam.debug_info:
                     debug_info.add_feature_error(beam.debug_info)
-            else:
+            except Warning as w:
```

Comment:
where is a `Warning` thrown?

Reply:
I never really used `try except`. I changed this to `FeatureApplicationError`, but I'm not sure if that is correct either.

Reply:
Ok the `beam.geometry` returns the `beam.blank`anyways, so that was redundant anyways. deleted the `try/except`